### PR TITLE
Dbarrosop/simplifications

### PR DESCRIPTION
**Pull request by dbarrosop**
Changes:

- Move `mandatory` elements, previously on the default action, to a dedicated action
- `from` is optional, by default it will always follow the `parent`
- `from` is now a pointer, no need to keep serializing/deserializing
- `mode` is optional. All parsers have a main `default` action now.
- On the JSONParser side of things, which was recently introduced by @bewing recently:
    - it's been refactored so it looks more like the XMLParser. At some point I'd actually get rid of the XMLParser and use only the JSONParser. The XMLParser would just convert initially the XML object into a dictionary and have the JSONParser deal with it
    - I got rid of some parsers because there were no tests and they are certainly broken after my changes
    - Things like subinterfaces are now supported.

Pinging @matejv @bewing @mirceaulinic @ktbyers 

The idea is to start getting rid of redundant things and make things as consistent and easy as possible. Simplification are far from complete so feel free to propose other changes that you think might be worth exploring.

### CODE REVIEW

File: napalm_yang/mappings/eos/parsers/state/openconfig-interfaces/interfaces.yaml [hunk 1]
```diff
@@ -27,82 +28,68 @@ interfaces:
             admin-status:
                 _process:
                     - mode: map
-                      value: "{{ bookmarks.parent.get('interfaceStatus', 'disabled') }}"
+                      path: interfaceStatus
+                      defaul: disabled
```
Comment (bewing) [inline napalm_yang/mappings/eos/parsers/state/openconfig-interfaces/interfaces.yaml]: Guessing this is a typo

File: test/integration/profiles_data/eos/openconfig-interfaces/default/openconfig-interfaces.native [hunk 2]
```diff
@@ -1 +1,323 @@
...ate": "unknown", "description": "", "bandwidth": 0, "mtu": 1500, "hardware": "ethernet", "duplex": "duplexFull", "loopbackMode": "loopbackNone", "physicalAddress": "08:00:27:66:ef:40", "interfaceCounters": {"outBroadcastPkts": 0, "linkStatusChanges": 1, "lastClear": 1497732401.4158947, "inMulticastPkts": 97, "counterRefreshTime": 1497734297.214895, "inBroadcastPkts": 2630, "inputErrorsDetail": {"runtFrames": 0, "fcsErrors": 0, "alignmentErrors": 0, "rxPause": 0, "symbolErrors": 0, "giantFrames": 0}, "inOctets": 212355, "outDiscards": 0, "outOctets": 43559, "inUcastPkts": 118, "outputErrorsDetail": {"lateCollisions": 0, "deferredTransmissions": 0, "txPause": 0, "collisions": 0}, "outUcastPkts": 104, "outMulticastPkts": 73, "totalInErrors": 0, "inDiscards": 0, "totalOutErrors": 0}, "interfaceStatistics": {"inBitsRate": 0.0, "updateInterval": 300.0, "outBitsRate": 0.0, "outPktsRate": 0.0, "inPktsRate": 0.0}, "interfaceAddress": [{"secondaryIpsOrderedList": [], "primaryIp": {"maskLen": 24, "address": "192.168.50.11"}, "broadcastAddress": "255.255.255.255", "secondaryIps": {}, "virtualIp": {"maskLen": 0, "address": "0.0.0.0"}}], "lineProtocolStatus": "up", "forwardingModel": "routed"}}}
```
Comment (bewing) [inline test/integration/profiles_data/eos/openconfig-interfaces/default/openconfig-interfaces.native]: I think we need to have further discussion regarding the test cases, especially as we start having more complex models that may have multiple commands.  As it stands, the single .native file is read as a string, as passed as the complete string being sole list entry in `native`.  How would we make this work when dealing with mappings that require multiple command execution?

Not saying we have to solve this now, just that it's still on my mind.

**Comment by bewing** [comment]
Will do some performance testing of oc-platform and my _oc_platform() method tomorrow.

File: napalm_yang/mappings/eos/parsers/state/openconfig-interfaces/interfaces.yaml (hunk 1 above)
Comment (dbarrosop) [inline napalm_yang/mappings/eos/parsers/state/openconfig-interfaces/interfaces.yaml]: Yes, thanks! I am actually going to remove the `default` argument. I don't think it's needed.

File: test/integration/profiles_data/eos/openconfig-interfaces/default/openconfig-interfaces.native (hunk 2 above)
Comment (dbarrosop) [inline test/integration/profiles_data/eos/openconfig-interfaces/default/openconfig-interfaces.native]: +1 to that. An easy solution would be to name the file `yadda.native.0, yadda.native.1` and use the number to place them in the correct position in the list.

**Pull request merged**